### PR TITLE
Support startup grace period and improve logging

### PR DESCRIPTION
**Pull request by asilvas-godaddy**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 coverage*
 node_modules/
 .idea
+package-lock.json
```
[the remark this answers is not in the document]
Comment (asilvas-godaddy) [reply .gitignore:4]: For packages I prefer not to lock dependencies.

File: lib/core/Happy.js [hunk 2]
```diff
@@ -15,12 +15,15 @@ module.exports = class Happy extends EventEmitter {
   constructor(opts) {
     super();
     this.options = Object.assign({
-      escalationSoftLimitMin: 20, // 20sec
-      escalationSoftLimitMax: 300, // 5min
+      escalationSoftLimitMin: 60, // 1min
+      escalationSoftLimitMax: 600, // 10min
```
Comment (jpage-godaddy) [inline lib/core/Happy.js:19]: Is this going to be a major bump since the defaults are changing?

Comment (asilvas-godaddy) [reply lib/core/Happy.js:19]: yes

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (pkarnawat-godaddy) [reply .gitignore:4]: This is for sure an interesting topic, https://dev.to/gajus/stop-using-package-lock-json-or-yarn-lock-3ddi

I think this package is going to get consumed directly. The author of that article also suggests adding a lock file in this scenario,

https://dev.to/gajus/comment/fo53
> I am not advocating against using package-lock.json for programs designed to be consumed by the end consumer. I do suggest not to use lock files in packages that will be dependencies of other packages.

By not committing the lock file, you could potentially break the dev environment, can potentially increase CI build time, prefer implicit trust on dependency (and their dependencies) authors to perform the correct versioning, etc.

**Review by jpage-godaddy — APPROVED** [review]
Would be better if the changes were documented so an upgrader knows what changed, but the change itself looks good.

**Comment by asilvas-godaddy** [comment]
> Would be better if the changes were documented so an upgrader knows what changed, but the change itself looks good.

added

**Pull request merged**